### PR TITLE
Akoya E Models hwdb

### PR DESCRIPTION
**Pull request by cvoinf**
In reply to #14272 , https://github.com/systemd/systemd/issues/14272#issuecomment-563175072

### CODE REVIEW

**Comment by hadess** [comment]
Please squash the commits as appropriate, this isn't easily reviewable.

**Comment by cvoinf** [comment]
I'm sorry, I forgot how to squash the changes and don't have the time right now to read me into git. Please, can you do this for me? Thank you. I guess for you it's made in 2 minutes, I will have to read at least an hour and afterwards it's perhaps wrong again.

To explain, what my suggestion is about:
When I bought these two laptops, I first was disappointed that Linux did not rotate the screen automatically. But after a while I found out how to change the hwdb so that my laptops were correctly recognized by systemd. 
What I realized was that all Akoya models beginning with Exxxx (up to now) have the same Matrix. So I propose to change the file 60-sensor.hwdb that all other Akoya E-models get the same matrix, since it is more likely that other models beginning with an E have the same matrix than that they don't have this one but the one (out of many possible permutations of that matrix) that is chosen if their exact fingerprint is not found in the list.

Alternatively you could add the lines for my two laptops:

`# Medion Akoya E3222`
`sensor:modalias:acpi:KIOX010A*:dmi:*:svnMEDION:pnMEDION*:*`
`ACCEL_MOUNT_MATRIX=0, -1, 0; -1, 0, 0; 0, 0, 1`

And the other one:

`# Medion Akoya E2293`
`sensor:modalias:acpi:KIOX010A*:dmi:*:svnMEDION:pnE2293MD61144*:*`
`ACCEL_MOUNT_MATRIX=0, -1, 0; -1, 0, 0; 0, 0, 1`


But with my first proposal, perhaps many owners of other E-models not in the list up to now are made happy; at least the probability is higher for that than for the possibility that somebody gets harmed.

**Comment by keszybz** [comment]
> Please squash the commits as appropriate, this isn't easily reviewable.

There's always the "Files changed" tab that shows a single diff for all commits. And for merging, "squash and merge" DTRT. So for commits like this, that should be just one commit, we don't need to ask contributors to squash. OTOH, when the PR is larger and we want to have multiple commits, then asking the contributor to rebase makes sense.

> What I realized was that all Akoya models beginning with Exxxx (up to now) have the same Matrix. So I propose to change the file 60-sensor.hwdb that all other Akoya E-models get the same matrix

That is harder to judge. @hadess what do you think?

**Comment by hadess** [comment]
> > Please squash the commits as appropriate, this isn't easily reviewable.
> 
> There's always the "Files changed" tab that shows a single diff for all commits. And for merging, "squash and merge" DTRT. So for commits like this, that should be just one commit, we don't need to ask contributors to squash. OTOH, when the PR is larger and we want to have multiple commits, then asking the contributor to rebase makes sense.

My time to work on this is limited. If you have time to try and untangle this, make sense of the changes, write a new commit message, then please go for it. I think that if the OP doesn't care to spend the time, maybe you can. I can't.

**Comment by keszybz** [comment]
OK, let's merge this then. If it turns out that the assumption that other Medion Akoya models are dissimilar, we will have to revert.

File: hwdb.d/60-sensor.hwdb [hunk 1]
```diff
@@ -433,27 +433,18 @@ sensor:modalias:acpi:SMO8500*:dmi:*:svnMEDION:pnAkoyaE2212TMD99720:*
...
+# and presumably all other Akoya Exxxx
 
-# Medion Akoya E2292 MD63390
-sensor:modalias:acpi:KIOX010A*:dmi:*:svnMEDION:pnE2292MD63390:*
+sensor:modalias:acpi:KIOX010A*:dmi:*:svnMEDION:pnE*:*
  ACCEL_MOUNT_MATRIX=0, -1, 0; -1, 0, 0; 0, 0, 1
 
-# Medion Akoya E2293 MD61130
-sensor:modalias:acpi:KIOX010A*:dmi:*:svnMEDION:pnE2293MD61130:*
+sensor:modalias:acpi:KIOX010A*:dmi:*:svnMEDION:pnMEDION*:*
```
Comment (keszybz) [inline hwdb.d/60-sensor.hwdb:445]: It would be nice to have the comment here too.

**Comment by cvoinf** [comment]
Thank you, guys!

From keszybz I got an email, saying it would be nice to comment the line 

> +sensor:modalias:acpi:KIOX010A*:dmi:*:svnMEDION:pnMEDION*:*

I'm sorry not to be so familiar with github that I know where to comment it best (perhaps in the source code, but that would mean we have to restart this procedure again...), so I'll try it here.





One Laptop is named Akoya E2293 MD61144. It's output is:

> cat  /sys/class/dmi/id/modalias

> dmi:bvnAmericanMegatrendsInc.:bvrGeminiLake_YS11G_V1.0.21:bd10/10/2018:svnMEDION:pnE2293MD61144:pvrELAN:rvnMEDION:rnYS11G:rvrDefaultstring:cvnMEDION:ct31:cvrDefaultstring:

The other one is named Akoya E3222 MD62450. It's output is:

> cat  /sys/class/dmi/id/modalias

> dmi:bvnAmericanMegatrendsInc.:bvrGeminiLake_YS13G_V1.0.21:bd10/10/2018:svnMEDION:pnMEDION:pvrNONE:rvnMEDION:rnMEDION:rvrDefaultstring:cvnMEDION:ct10:cvrDefaultstring:

As you can see, the last output has this svnMEDION:pnMEDION -string inside. It has the same Matrix.


Thank's again for helping!

**Comment by keszybz** [comment]
@cvoinf thanks. I added that comment when squashing your PR, so it's all fine now.

**Comment by poettering** [comment]
Hmm, seems this wasn't entirely correct, see #14437

**Comment by hadess** [comment]
> Hmm, seems this wasn't entirely correct, see #14437

Narrator: it was correct.

**Pull request merged**